### PR TITLE
More Menubar

### LINKED ISSUE #1294: Client Settings tab Key Bindings is a black hole when using a keyboard
### Environment
What version of MegaMek does your issue apply to?  0.45.X
What operating system are you using?  Windows
What java version are you using?  8

### Description
If you're navigating with a keyboard, you get stuck once you tab through to the Key Bindings tab under Client Settings. The first "modifiers" text box takes focus and captures keys like TAB.

### PR DESCRIPTION
**Pull request by SJuliez**
The last of bigger changes to the menu bar:
- Splits the former File menu into Game and Units menus
- Removes all in-game actions (such as Turn Left or Accelerate) (!) as this felt really triassic. These are and should be covered my mouse actions and the button bar
- Adds the menu accelerators to the KeyCommandBinds. This makes the menu accelerators fully configurable in the keybinds section of the Settings. 
- Adds a few menu items and some more accelerators
- Cleans up the enabled/disabled states of the menu items 
- The only exceptions are Copy/Paste in the lobby which use hard-coded Ctrl-C/V as these seem quite universal and I had to spend a little more effort in catching them as tables eat them
- Changes to keybinds now take effect immediately (after closing the Settings dialog)
- Removes the "repeatable" option from the Settings Dialog/Keybinds. Being repeatable is just what a command does and makes sense only for the scroll commands. No use in making it configurable.
- The Settings Dialog/Keybinds will now warn about duplicate keybinds for menu keybinds and "exclusive" keybinds (e.g. for Done). (This does not cover all possibilities. When there's no warning things might still not work well) Also, entering modifiers like Shift in the "key" part of keybind (instead of the modifier part) is prevented
- Corrects what I think is an error in the KeyBindParser where the hard-coded default keys would only be loaded if the xml file was missing. This led e.g. to the keybinds overlay hotkey not working for most people. Instead the hard-coded defaults are now always loaded and possibly overridden by those from the defaultKeyBinds.xml. The keybinds should now handle better if people copy in their own xml or if a keybind is added without updating the xml.
- The unit display will now behave like the minimap and unit overview in that it's a user preference if it will be shown automatically at phase changes. If it's turned off, it'll stay off until activated again (which can be done via keybind, menubar or popup menu "View <Unit>")
- Removes some GUIPref settings from the advanced settings (hex coords and toggle unit labels)
Resolves #1294

image image image

### CODE REVIEW

**Comment by sixlettervariables** [comment]
Some things I have concerns with:
1. Removing some of the movement related actions. We need to find out if our Accessibility community uses those. @BlindGuyNW 
2. The Game menu should have regular accelerators for Save, Open, and Exit or Quit. (E.g. ALT-G O for Save). I'd really prefer if it were still File, but perhaps that's not such a big deal.

**Comment by SJuliez** [comment]
Re 1: OK, when they're needed, they're needed. I don't have a quick solution that would make the buttons equally accessible. But I do think I'd add a settings switch to hide them.

Re 2: I can take a look at the mnemonics. I don't recall actively removing any, at least not on purpose, but I can sure add some. I could make it a File menu and change Units -> Game and move over things like the Round Report from File to Game. Np. Better do it now. Opinions?

**Comment by sixlettervariables** [comment]
Yes for (2) if we don't have mnemonics for the standard menu items, we probably should (separate from accelerators).

https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/swing/AbstractButton.html#setMnemonic(int)

**Comment by SJuliez** [comment]
@sixlettervariables I just noticed that the buttons in the action button bar are actually reachable by tabbing, including the Done button. I'd say these are far more accessible than setting a move target hex or choosing a weapon to fire. This makes me think we can safely remove the menu entries. However I also changed a few default keybinds and among them used TAB as it's a real waste not to use it (for Next Unit). The keybind screen is really inaccesible though as it doesn't allow tabbing in order to be able to set TAB as a keybind. So it's hard to get rid of the TAB entry if you're restricted to tabbing in the client settings. I can imagine adding a toggle at the top of that dialog tab (why is everything called tab) that will allow tabbing (and thus, prevent TAB from being added as a keybind); or even actually remove TAB from any keybind immediately.

**Comment by SJuliez** [comment]
@sixlettervariables Added a toggle to the keybinds to allow tabbing. As I wrote above enabling this will remove all TABs from existing keybinds as I assume that people using this will rather want TAB to not be bound. I also changed the key textfields to not react at all when modifiers are held so the Update/Cancel mnemonics can be used at least when in a key textfield. In a modifier textfield that doesn't work of course.

image

**Comment by SJuliez** [comment]
Ah, and also changed Game -> File and Units -> Game and moved over the round report etc. menu items from now File to now Game

**Comment by Windchild292** [comment]
@SJuliez
This has merge conflicts

**Review by Windchild292 — COMMENTED** [review]
Just a few things from review

File: megamek/src/megamek/client/ui/swing/lobby/ChatLounge.java [hunk 1]
```diff
@@ -3560,5 +3558,18 @@ boolean isForceView() {
         return butForceView.isSelected();
     }
     
+    /** Returns true when a modal dialog such as the Camo Chooser or a Load Force dialog is currently shown. */
+    public boolean isModalDialogShowing()
+    {
+        Window[] windows = Window.getWindows();
+        if (windows != null) {
+            for (Window w : windows) {
+                if (w.isShowing() && (w instanceof JDialog) && ((JDialog)w).isModal())
```
[the remark this answers is not in the document]
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/lobby/ChatLounge.java:3567]: Stream.of isnt null-safe but getWindows doesnt return null so indeed a stream is ok here. Will change it.

**Comment by SJuliez** [comment]
OK, all done.

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**